templates/mysqlbackup.sh.erb

```diff
@@ -16,9 +16,18 @@ DIR=<%= @backupdir %>
 DAYS=<%= @backupdays %>
 
 ##### STOP CONFIG ####################################################
+MYSQLDUMP="mysqldump -u${USER} -p${PASS} --opt --flush-logs --single-transaction"
+DATE=`date +%Y%m%d-%H%M%S`
 PATH=/usr/bin:/usr/sbin:/bin:/sbin
 
-find $DIR -mtime +${DAYS} -exec rm -f {} \;
-mysqldump -u${USER} -p${PASS} --opt --flush-logs --single-transaction \
- --all-databases <% if @backupcompress %>| bzcat -zc <% end %>> ${DIR}/mysql_backup_`date +%Y%m%d-%H%M%S`.sql<% if @backupcompress %>.bz2<% end  %>
+find $DIR -type f -mtime +${DAYS} -delete
 
+<%- if @onefile -%>
+FNAME="${DIR}/mysql_backup_${DATE}.sql<% if @backupcompress %>.bz2<% end  %>"
+${MYSQLDUMP} --all-databases <% if @backupcompress %>| bzcat -zc <% end %>> ${FNAME}
+<%- else -%>
+for DB in $( echo "show databases" |mysql -N ${COPTS} |grep -v '^#'); do
```

Comment:
You'll need to change the mysql command to act like the mysqldump command here I think to pass in the credentials. Also what is ${COPTS}